src/jshint.js

```diff
@@ -2015,7 +2026,7 @@ var JSHINT = (function() {
               funct[v] = s["(global)"] ? "global" : "outer";
               break;
             case "const":
-              setprop(s, v, { unused: false });
+              s[v]["(type)"] = false;
```

Comment:
the code being replaced looks related to legacy const stuff, not sure. In general though, I wouldn't expect you to need to do anything to blockscope `const` where similar work wasn't already done for `let`. The two should be pretty consistent, just slightly different semantics for `const`.